### PR TITLE
Check for deprecated functions in tutorials

### LINKED ISSUE #10566: Warn about deprecated stuff in the tutorials
#10565 shows that we don't get compiler warnings when we use deprecated functions in the tutorial programs. That seems to be an oversight: We really shouldn't be using deprecated things in the tutorial programs.

What would it take to change the compiler flags so that tutorial programs aren't treated like internal source files?

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #10566.
For some reason, `clang-format` messes up the indentation in `include/deal.II/matrix_free/matrix_free.h` so I had to disable it.

### CODE REVIEW

**Comment by tjhei** [comment]
> For some reason, `clang-format` messes up the indentation in `include/deal.II/matrix_free/matrix_free.h` so I had to disable it.

can you try DEAL_II_DISABLE_EXTRA_DIAGNOSTICS instead?

**Comment by masterleinad** [comment]
> can you try DEAL_II_DISABLE_EXTRA_DIAGNOSTICS instead?

I expect that to work but oit's a little bit of a sledgehammer approach. :slightly_smiling_face: I don't feel that strongly about it and pushed a corresponding commit.

**Comment by tjhei** [comment]
> a little bit of a sledgehammer approach.

agreed, but better than ``clang-format off``, I feel.

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -48,6 +48,11 @@ IF(DEAL_II_COMPONENT_EXAMPLES)
     )
 
   IF(DEAL_II_COMPILE_EXAMPLES)
+    #
+    # Make sure that there are no deprecated functions used in the tutorials.
+    #
+    STRIP_FLAG(DEAL_II_CXX_FLAGS "-Wno-deprecated-declarations")
```
Comment (bangerth) [inline examples/CMakeLists.txt:54]: Nice -- thanks!

**Pull request merged**